cpp/src/slice2java/Gen.cpp

```diff
@@ -4313,93 +4313,37 @@ Slice::Gen::ProxyVisitor::visitInterfaceDefEnd(const InterfaceDefPtr& p)
         << p->name() << "PrxI.class);";
     out << eb;
 
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_context(java.util.Map<String, String> newContext);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_adapterId(String newAdapterId);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_endpoints(com.zeroc.Ice.Endpoint[] newEndpoints);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_locatorCacheTimeout(int newTimeout);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_invocationTimeout(int newTimeout);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_connectionCached(boolean newCache);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_endpointSelection(com.zeroc.Ice.EndpointSelectionType newType);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_secure(boolean b);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_encodingVersion(com.zeroc.Ice.EncodingVersion e);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_preferSecure(boolean b);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_router(com.zeroc.Ice.RouterPrx router);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_locator(com.zeroc.Ice.LocatorPrx locator);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_collocationOptimized(boolean b);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_twoway();";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_oneway();";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_batchOneway();";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_datagram();";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_batchDatagram();";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_compress(boolean co);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_timeout(int t);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_connectionId(String connectionId);";
-
-    out << sp;
-    out << nl << "@Override";
-    out << nl << p->name() << "Prx ice_fixed(com.zeroc.Ice.Connection connection);";
+    // Generate overrides for all the methods on `ObjectPrx` with covariant return types.
+    static const string objectPrxMethods[] = {
```

Comment:
I would use constexpr and string_view here.

Reply:
Fixed!